packages/components/atoms/f-form-field/src/assets/scss/common.scss

```diff
@@ -22,31 +22,45 @@ $form-input-borderColour--invalid       : $color-support-error;
 
 $form-input-focus--boxShadow            : 0 0 0 2px $color-focus;
 
-$form-input-height-small                : spacing(x5);
-$form-input-padding-small               : spacing(x1) spacing(x2);
+$form-input-height--small                : spacing(x5);
+$form-input-padding--small               : spacing(x1) spacing(x2);
 
-$form-input-height-medium               : spacing(x6);
-$form-input-padding-medium              : spacing(x1.5) spacing(x2);
+$form-input-height--medium               : spacing(x6);
+$form-input-padding--medium              : spacing(x1.5) spacing(x2);
 
-$form-input-height-large                : spacing(x7);
-$form-input-padding-large               : spacing(x2);
+$form-input-height--large                : spacing(x7);
+$form-input-padding--large               : spacing(x2);
 
+$form-input-iconPadding                 : spacing(x7);
+$form-input-iconPosition                : 19px;
 
 @mixin form-field-size($height, $padding) {
     @include rem(height, $height);
     padding: $padding;
 }
 
+@mixin icon-position($size, $position: 'bottom') {
+    svg {
+        @if $position == 'leading' {
+            left: $form-input-iconPosition;
```

Comment:
Just looking at how this gets it's values, it's a little bit confusing atm. It takes a param of `size`, but for leading and trailing, ignores the `size` param and just uses the outer scoped `$form-input-iconPosition` (which is the same as the `size` passed in?).

Also, it looks like in some instances (when bottom is set) this helps control the size of the icon, but when leading/trailing is set, then size isn't needed at all?

May be worth catching up on hangout, as I think you can likely simplify this a little. I think either the size param needs renaming (and then you can use it when trailing or leading position is set) or need to rejig it so that size becomes an optional parameter and just use the globally set variables (so don't bother passing them in for leading or trailing icons).

Reply:
That's way better, thanks Ash